### DISCUSSION TITLE
Optionally persist variables and time range across dashboards

### DISCUSSION THREAD

**Discussion by jmcarp** [grafana/grafana#59409] [Dashboards, Panels & Visualizations]
It would be useful to have the option to make variable selections and time ranges persistent from one dashboard to the next. For example, if I select a given time range on one dashboard and then navigate to another dashboard, I want the time range from the first to be selected for the second. Similarly, I select the "production" environment while browsing one dashboard, I want the same environment to be selected when I navigate to a second dashboard. I'm guessing this wouldn't be too complicated to implement by appending variable and time range selections to url query strings.

Note: I might be able to solve this problem with dashboard links, but I don't want each dashboard to have to enumerate all possible dashboards I might navigate to next—I want to be able to opt in to this behavior for all dashboards.

**Comment by Kenterfie** [comment]
In general, it would not be bad to be able to use the timestamp across multiple dashboards. When searching for a problem, you look explicitly at a time range and it would be advantageous not to have to specify it again. Copying from/to only helps to a limited extent. I would also very much welcome such a function.

**Reply by Kenterfie** [reply]
One idea is to have a lock sign next to the time range selection which allows the user to lock the current time range across multiple dashboards.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
